src/Button/RedliningButton/RedliningButton.jsx

```diff
@@ -0,0 +1,672 @@
+        }),
+        style: this.getRedliningStyleFunction
+      });
+      map.addLayer(redliningLayer);
+    }
+    this.setState({redliningLayer});
+  }
+
+  /**
+   * The styling function for the redlining vector layer, which considers
+   * different geometry types of drawn features.
+   *
+   * @param {OlFeature} feature The feature which is being styled.
+   * @return {OlStyleStyle} The style to use.
+   *
+   * @method
+   */
+  getRedliningStyleFunction = feature => {
+
+    const {
+      fillColor,
+      selectFillColor,
+      strokeColor,
+      selectStrokeColor
+    } = this.props;
+
+    const {
+      interaction,
+    } = this.state;
+
+    let useSelectStyle = false;
+    if (interaction && interaction.length > 1 && !feature.get('isLabel')) {
+      useSelectStyle = true;
+    }
+
+    switch (feature.getGeometry().getType()) {
+      case 'Point': {
+        if (!feature.get('isLabel')) {
+          return new OlStyleStyle({
+            image: new OlStyleCircle({
+              radius: 7,
+              fill: new OlStyleFill({
+                color: !useSelectStyle ? fillColor : selectFillColor
+              }),
+              stroke: new OlStyleStroke({
+                color: !useSelectStyle ? strokeColor : selectStrokeColor
+              })
+            })
+          });
+        } else {
+          return new OlStyleStyle({
+            text: new OlStyleText({
+              text: '',
+              offsetX: 5,
+              offsetY: 5,
+              font: '12px sans-serif',
+              fill: new OlStyleFill({
+                color: !useSelectStyle ? fillColor : selectFillColor
+              }),
+              stroke: new OlStyleStroke({
+                color: !useSelectStyle ? strokeColor : selectStrokeColor
+              })
+            })
+          });
+        }
+      }
+      case 'LineString': {
+        return new OlStyleStyle({
+          stroke: new OlStyleStroke({
+            color: !useSelectStyle ? strokeColor : selectStrokeColor,
+            width: 2
+          })
+        });
+      }
+      case 'Polygon':
+      case 'Circle': {
+        return new OlStyleStyle({
+          fill: new OlStyleFill({
+            color: !useSelectStyle ? fillColor : selectFillColor
+          }),
+          stroke: new OlStyleStroke({
+            color: !useSelectStyle ? strokeColor : selectStrokeColor,
+            width: 2
+          })
+        });
+      }
+      default: {
+        break;
+      }
+    }
+  }
+
+  /**
+   * Creates a correctly configured OL draw interaction depending on given
+   * drawType and adds this to the map.
+   *
+   * @param {Boolean} pressed Whether the redlining button is pressed or not.
+   * Will be used to handle active state of the draw interaction.
+   *
+   * @method
+   */
+  createDrawInteraction = pressed => {
+    const {
+      drawType,
+      map
+    } = this.props;
+
+    let geometryFunction;
+    let type = drawType;
+
+    if (drawType === 'Rectangle') {
```

Comment:
Minor: Please move all draw type strings to statics.

Reply:
fixed in df4e3fd